hermes/probe/create/create.go

```diff
@@ -0,0 +1,186 @@
+//          fileID: the unique identifer of every randomFile, it cannot be repeated. It needs to be in the range [begin, end]. FileID 0 is reserved for a special file called the NIL file.
+//          client: is a storage client. It is used as an interface to interact with the target storage system.
+//          logger: a cloudprober logger used to record the exit status of the CreateFile operation in a target bucket. The logger passed MUST be a valid logger.
+// Returns:
+//          error: an error string with detailed information about the status and fileID. Nil is returned when the operation is successful.
+func CreateFile(ctx context.Context, target *probe.Target, fileID int32, fileSize int, client stiface.Client, logger *logger.Logger) error {
+	f, err := newRandomFile(fileID, fileSize)
+	if err != nil {
+		return err
+	}
+	fileName, err := f.fileName()
+	if err != nil {
+		return err
+	}
+	target.Journal.Intent = &pb.Intent{FileOperation: pb.Intent_CREATE, Filename: fileName}
+	var status metrics.ExitStatus
+	if _, ok := target.Journal.Filenames[fileID]; ok {
+		status = metrics.UnknownFileFound
+		return fmt.Errorf("CreateFile(ID: %d).%q could not create file as file with this ID already exists", fileID, status)
+	}
+	r := f.NewReader()
+	start := time.Now()
+	bucketName := target.Target.GetBucketName()
+	wc := client.Bucket(bucketName).Object(fileName).NewWriter(ctx)
+	if _, err = io.Copy(wc, r); err != nil {
+		switch err {
+		case storage.ErrBucketNotExist:
+			status = metrics.BucketMissing
+		default:
+			status = metrics.ProbeFailed
+		}
+		target.LatencyMetrics.APICallLatency[metrics.APICreateFile][status].Metric(hermesAPILatencySeconds).AddFloat64(time.Now().Sub(start).Seconds())
+		return fmt.Errorf("CreateFile(id: %d).%q: could not create file %q: %v", fileID, status, fileName, err)
+	}
+	if err := wc.Close(); err != nil {
+		status = metrics.WriterCloseFailed
+		target.LatencyMetrics.APICallLatency[metrics.APICreateFile][status].Metric(hermesAPILatencySeconds).AddFloat64(time.Now().Sub(start).Seconds())
+		return fmt.Errorf("Writer.Close: %v with status %q", err, status)
+	}
+	status = metrics.Success
+	target.LatencyMetrics.APICallLatency[metrics.APICreateFile][status].Metric(hermesAPILatencySeconds).AddFloat64(time.Now().Sub(start).Seconds())
+
+	// Verify that the file that has just been created is in fact present in the target system
+	query := &storage.Query{Prefix: fileName[:FileNamePrefixSize]}
+	start = time.Now()
+	objIter := client.Bucket(bucketName).Objects(ctx, query)
+	finish := time.Now()
+	obj, err := objIter.Next()
+	if err != nil {
+		target.LatencyMetrics.APICallLatency[metrics.APIListFiles][metrics.FileMissing].Metric(hermesAPILatencySeconds).AddFloat64(finish.Sub(start).Seconds())
+		return fmt.Errorf("CreateFile check failed: %v", err)
+	}
+	if obj.Name != fileName {
```

Comment:
This assumes that only a single result is returned by the query.
What if there are multiple results?

Reply:
we check whether there was a file with this ID in the logger at the beginning of the create function and also we check whether a file with this ID exists on the storage system by calling the Read probe and if we call create after deletion there is a check performed at the end of deletion to see whether a file with this ID does not exist on the storage sysem so essentially if there are multiple results it would mean that we created more than one file during this probe run?

Therefore should we consider the case with the multiple results and if so should this then be changed to 
for {
} with a break statement if obj.Next = nil? 

or if there are multiple results  it clearly means that something went wrong so just return an error?

Reply:
I would use a for loop and also error out if more than one file is found (listing all the files found in the error).

Reply:
changed to a loop and errored out if a list is found and also returning the list of files found